src/Build/BackEnd/Shared/BuildResult.cs

```diff
@@ -433,6 +437,17 @@ internal List<string> DefaultTargets
             { _defaultTargets = value; }
         }
 
+        /// <summary>
+        /// The defined targets for the project associated with this build result.
+        /// </summary>
+        public HashSet<string> ProjectTargets
```

Comment:
This is adding to the public API. Is this needed?

Reply:
Huh, looks like I can get away with it making `internal`—no need to add it to the public API.

I do believe it's functionally needed, for reasons described in `BuildManager.HandleResult`. If the `BuildManager` doesn't have the `ProjectTargets` for a particular configuration, the scheduler (which uses the same global config cache as the `BuildManager`) won't be able to `SkipNonexistentTargets` properly.